app/src/engineering/java/com/hedvig/app/feature/embark/EmbarkStoryTesterActivity.kt

```diff
@@ -0,0 +1,259 @@
+package com.hedvig.app.feature.embark
+
+import android.os.Bundle
+import androidx.activity.compose.setContent
+import androidx.appcompat.app.AppCompatActivity
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.LazyRow
+import androidx.compose.foundation.lazy.items
+import androidx.compose.material.Button
+import androidx.compose.material.Icon
+import androidx.compose.material.IconButton
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Surface
+import androidx.compose.material.Text
+import androidx.compose.material.TextField
+import androidx.compose.material.TopAppBar
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.ArrowBack
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.google.accompanist.insets.systemBarsPadding
+import com.hedvig.app.authenticate.AuthenticationTokenService
+import com.hedvig.app.feature.embark.ui.EmbarkActivity
+import com.hedvig.app.feature.onboarding.BundlesResult
+import com.hedvig.app.feature.onboarding.GetBundlesUseCase
+import com.hedvig.app.feature.settings.Market
+import com.hedvig.app.feature.settings.MarketManager
+import com.hedvig.app.ui.compose.theme.HedvigTheme
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import org.koin.androidx.viewmodel.dsl.viewModel
+import org.koin.androidx.viewmodel.ext.android.viewModel
+import org.koin.dsl.module
+
+class EmbarkStoryTesterActivity : AppCompatActivity() {
+
+    val model: EmbarkStoryTesterViewModel by viewModel()
+
+    override fun onResume() {
+        super.onResume()
+        model.onStorySelected(null)
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        setContent {
+            val viewState by model.viewState.collectAsState()
+
+            viewState.selectedStoryName?.let {
```

Comment:
This will potentially trigger multiple times on configuration change, right? 🤔 I think this value might need to be cleared after the 'event' has been consumed, in order to mitigate that issue.

Reply:
Yeah that is correct, doing such a side effect from inside Compose can very easily trigger more than once if this function recomposes again, and there may be various reasons why it may recompose.
I'd say something like this should work fine:
```kotlin
LaunchedEffect(viewState.selectedStoryName) {
    val selectedStoryName = viewState.selectedStoryName ?: return@LaunchedEffect
    model.onStorySelected(null)
    startActivity(
        EmbarkActivity.newInstance(
            this@EmbarkStoryTesterActivity,
            selectedStoryName,
            selectedStoryName,
        )
    )
}
```
This makes sure it only runs every time `selectedStoryName` changes, and it does nothing when it's null, and it clears the value before it launches the activity so that if you come back it doesn't trigger it again by any chance.

Should work I think, didn't run it myself

Reply:
Thanks